modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/display/context/CommercePriceListDisplayContext.java

```diff
@@ -260,20 +260,42 @@ public long getParentCommercePriceListId() throws PortalException {
 		return commercePriceList.getParentCommercePriceListId();
 	}
 
-	public CreationMenu getPriceListCreationMenu() throws Exception {
+	public CreationMenu getPriceListCreationMenu(String portletName)
+		throws Exception {
+
 		CreationMenu creationMenu = new CreationMenu();
 
+		String title = StringPool.BLANK;
+
 		if (hasPermission(
 				CommercePriceListActionKeys.ADD_COMMERCE_PRICE_LIST)) {
 
-			creationMenu.addDropdownItem(
-				dropdownItem -> {
-					dropdownItem.setHref(getAddCommercePriceListRenderURL());
-					dropdownItem.setLabel(
-						LanguageUtil.get(
-							httpServletRequest, "create-new-price-list"));
-					dropdownItem.setTarget("modal-lg");
-				});
+			if (portletName.equals(
+					CommercePricingPortletKeys.COMMERCE_PRICE_LIST)) {
+
+				creationMenu.addDropdownItem(
+					dropdownItem -> {
+						dropdownItem.setHref(
+							getAddCommercePriceListRenderURL());
+						dropdownItem.setLabel(
+							LanguageUtil.get(
+								httpServletRequest, "create-new-price-list"));
+						dropdownItem.setTarget("modal-lg");
+					});
+			}
+			else if (portletName.equals(
+						CommercePricingPortletKeys.COMMERCE_PROMOTION)) {
+
+				creationMenu.addDropdownItem(
+					dropdownItem -> {
+						dropdownItem.setHref(
+							getAddCommercePriceListRenderURL());
+						dropdownItem.setLabel(
+							LanguageUtil.get(
+								httpServletRequest, "create-new-promotion"));
+						dropdownItem.setTarget("modal-lg");
+					});
+			}
```

Comment:
I think you can reassign the variable instead of duplicating the code

Reply:
Yeah, that was my thought at the first moment and i tried different ways to do that, but unfortunately didn't work and the best option was to duplicate it. 
But if you have a suggestion of how i can reassign the variable i can try again.

Reply:
```
if (portletName.equals(
					CommercePricingPortletKeys.COMMERCE_PRICE_LIST)) {
					
					title = "create-new-price-list";
			}
			else if (portletName.equals(
						CommercePricingPortletKeys.COMMERCE_PROMOTION)) {
					
					title = "create-new-promotion";
			}
			
			

		creationMenu.addDropdownItem(
			dropdownItem -> {
				dropdownItem.setHref(
					getAddCommercePriceListRenderURL());
				dropdownItem.setLabel(
					LanguageUtil.get(
						httpServletRequest, title));
				dropdownItem.setTarget("modal-lg");
			});
```